### ISSUE TITLE
Add CLI support for easier use in terminal

### ISSUE THREAD

**Issue by gitbutler-client** [gitbutlerapp/gitbutler#2515]
Are there any plans for a CLI? I'm a heavy user of `git stash` and stage changes exclusively with `git add -p`. GitButler should fit right in and it would be great if leaving the terminal was optional.

via: @tt

https://discord.com/channels/1060193121130000425/1176875943873232947/1198631782011129906

**Comment by jsvensson** [comment]
In addition to this, opening a repo in GitButler from the terminal would be great, ie `gitbutler .` when in the repo in the terminal.

**Comment by rohitpaulk** [comment]
I remember seeing a comment around some CLI tool used that's internally for testing, can't find it at the moment. Was wondering if there's a way to send commands to the GitButler app. If a few simple commands like "commit changes to default lane", "create PR from default lane" etc. were exposed, this could allow users to experiment with creating their own CLI scripts.

**Comment by Byron** [comment]
Indeed, there is an internal tool that can easily be built from source with `cargo build --release --bin but-cli`, but it's purpose is to aid debugging and testing of the backend. It's not ~~'nice'~~ very 'practical' and naturally doesn't handle anything that is entirely done in the frontend right now, like PR management.
It will be a while, but it's definitely on the list of planned features.

**Comment by Caleb-T-Owens** [comment]
Hi! We're actively working on a CLI now. In our latest release we've introduced some key commands that cover a pretty large set of the UI functionality. Namely `rub`, `init`, and `commit`.

The CLI can be added to your path from the global settings page.

<img width="1548" height="575" alt="Image" src="https://github.com/user-attachments/assets/60a49022-8435-4a5d-92e3-c162aa3a9deb" />
